code/code/cmd/cmd_stab.cc

```diff
@@ -1,345 +1,239 @@
-      // Normal:  0 :  1 in 501
-
-      int tNewSever = (int)(((float)tSever * -tDamageValues[tLimb]) - 4.0);
-
-      // We further modify it based on the limb for a reason:
-      // Chance for whacking:
-      // Head   :  2 (can not be whacked)
-      // Neck   :  1 (can not be whacked)
-      // Body   :  1 (can not be whacked)
-      // Back   :  1 (can not be whacked)
-      // Arms   :  0 (can     be whacked)
-      // Waist :  2 (can not be whacked)
-      // Legs   :  1 (can not be whacked)
-      // Fingers: -4 (can     be whacked)
-      if (::number(tNewSever, 500) <= 0 &&
-          !tSucker->isLucky(tSucker->GetMaxLevel())) {
-        tStringChar = format("You slice $N's %s right off!") % tStLimb;
-        tStringVict = format("$n slices your %s right off!") % tStLimb;
-        tStringOthr="$n slices $N's %s right off!") % tStLimb;
-
-        tSucker->makePartMissing(tLimb, false, tThief);
-
-        act(tStringChar, FALSE, tThief, NULL, tSucker, TO_CHAR);
-        act(tStringVict, FALSE, tThief, NULL, tSucker, TO_VICT);
-        act(tStringOthr, FALSE, tThief, NULL, tSucker, TO_NOTVICT);
+      // New bleed
+      sstring limbName(victim->describeBodySlot(limb));
+      act(format("Your stab opens a <r>bleeding wound<z> on $N's %s!") % limbName,
+        FALSE, thief, nullptr, victim, TO_CHAR);
+      act(format("$n's stab opens a <r>bleeding wound<z> on your %s!") % limbName,
+        FALSE, thief, nullptr, victim, TO_VICT);
+      act(format("$n's stab opens a <r>bleeding wound<z> on $N's %s!") % limbName,
+        FALSE, thief, nullptr, victim, TO_NOTVICT);
+
+      victim->rawBleed(limb, duration, SILENT_YES, CHECK_IMMUNITY_NO);
+    }
+  }
+}
+
+// Offhand stab attack - uses SKILL_STABBING for specAttack
+static int stabOffAtk(TBeing* thief, TBeing* victim, TGenWeapon* weapon) {
+  if (!weapon || !weapon->canStab())
+    return FALSE;
+
+  // Use DEX/SPE for offense, AGI/PER for defense
+  int specResult = thief->specialAttack(victim, SKILL_STABBING, 0, STAT_DEX,
+    STAT_SPE, STAT_AGI, STAT_PER, false);
+  if (specResult != COMPLETE_SUCCESS && specResult != GUARANTEED_SUCCESS) {
+    stabMissMsg(thief, victim, weapon);
+    victim->addHated(thief);
+    return TRUE;
+  }
+
+  // Calculate damage for offhand
+  int level = thief->getSkillLevel(SKILL_STABBING);
+  int dam = thief->getSkillDam(victim, SKILL_STABBING, level,
+    thief->getAdvLearning(SKILL_STABBING));
+  dam = thief->getActualDamage(victim, weapon, dam, SKILL_STABBING);
+  
+  if (dam <= 0) {
+    act("Your $o fails to penetrate $N's thick hide.", FALSE, thief, weapon,
+      victim, TO_CHAR);
+    act("$n's $o fails to penetrate your thick hide.", FALSE, thief, weapon,
+      victim, TO_VICT);
+    act("$n's $o fails to penetrate $N's thick hide.", FALSE, thief, weapon,
+      victim, TO_NOTVICT);
+    return TRUE;
+  }
+
+  // Get target limb
+  wearSlotT limb = victim->getPartHit(thief, FALSE);
+  if (!victim->hasPart(limb))
```

Comment:
Why do other call sites of `getPartHit` not check for this? Is there a bug in other call sites? Is this check redundant?